solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -622,11 +630,8 @@ private void setBasicAuthHeader(SolrRequest<?> solrRequest, Request req) {
   private void decorateRequest(Request req, SolrRequest<?> solrRequest, boolean isAsync) {
     req.headers(headers -> headers.remove(HttpHeader.ACCEPT_ENCODING));
 
-    if (requestTimeoutMillis > 0) {
-      req.timeout(requestTimeoutMillis, TimeUnit.MILLISECONDS);
-    } else {
-      req.timeout(idleTimeoutMillis, TimeUnit.MILLISECONDS);
-    }
+    req.timeout(requestTimeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
We can & should keep this simple.  there are 2 clients sharing some foundation, and we now resolve a default early so here we just do something obvious.